### PR TITLE
Non-allocating XML parser.

### PR DESCRIPTION
**Pull request by katmsft**
Besides the draft, I still have trouble figuring out the best design for 'array' values in XML. Any thoughts?

### CODE REVIEW

File: sdk/core/core/inc/az_xml_parser.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  az_xml_stack stack;
+} az_xml_parser;
+
+AZ_NODISCARD az_xml_parser az_xml_parser_create(az_span const buffer);
+
+AZ_NODISCARD az_result
+az_xml_parser_read(az_xml_parser * const self, az_xml_value * const out_value);
+
+AZ_NODISCARD az_result
+az_xml_parser_read_object_member(az_xml_parser * const self, az_xml_member * const out_member);
```
Comment (sergey-shandar) [inline sdk/core/core/inc/az_xml_parser.h]: I think, XML should have different interface. Like `read_attribute`, `read_item` etc.

File: sdk/core/core/src/az_xml_parser.c [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+AZ_NODISCARD az_result az_xml_parser_skip(
+    az_xml_parser * const self,
+    az_xml_value const value
+)
+{
+    (void)self;
+
+    return AZ_ERROR_NOT_IMPLEMENTED;
+}
```
Comment (antkmsft) [inline sdk/core/core/src/az_xml_parser.c]:
```suggestion
}

```

File: sdk/core/core/inc/az_xml_value.h [hunk 3]
```diff
@@ -0,0 +1 @@
+#pragma once
```
Comment (antkmsft) [inline sdk/core/core/inc/az_xml_value.h]: Don't use `#pragma once` in your real implementation

File: sdk/core/core/src/az_xml_parser.c [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+AZ_NODISCARD az_result az_xml_parser_skip(
+    az_xml_parser * const self,
+    az_xml_value const value
+)
+{
+    (void)self;
```
Comment (antkmsft) [inline sdk/core/core/src/az_xml_parser.c]: AZ_CONTRACT_ARG_NOT_NULL()?

File: sdk/core/core/inc/az_xml_parser.h (hunk 1 above)
Comment (katmsft) [reply sdk/core/core/inc/az_xml_parser.h]: Yes, I was trying to implement a new design after the talk. Sorry I was occupied by other things. I'll be focusing on this from this week on.

File: sdk/core/core/src/az_xml_parser.c (hunk 2 above)
Comment (katmsft) [reply sdk/core/core/src/az_xml_parser.c]: Sure, thanks for the comment.

File: sdk/core/core/inc/az_xml_value.h (hunk 3 above)
Comment (katmsft) [reply sdk/core/core/inc/az_xml_value.h]: Cool, removed :)

File: sdk/core/core/src/az_xml_parser.c (hunk 4 above)
Comment (katmsft) [reply sdk/core/core/src/az_xml_parser.c]: Adding all those pre-condition check in the implementation.

**Comment by antkmsft** [comment]
It's been a long ago since this PR is in review. Should we close it?

**Pull request closed without merging**